src/utils/commonRequest.ts

```diff
@@ -19,24 +19,122 @@ export async function createPlayer(
   await client.query(query, values)
 }
 
-export async function isPlayerInGame(
+export async function isPlayerInAnyGame(
   playerToken: string,
   client: Client
 ): Promise<boolean> {
-  const query = 'SELECT * FROM Players WHERE token=$1'
+  const query = 'SELECT 1 FROM Players WHERE token=$1'
   return (await client.query(query, [playerToken])).rowCount !== 0
 }
 
+export async function isPlayerInGame(
+  playerToken: string,
+  gameId: string,
+  client: Client
+): Promise<boolean> {
+  const query = 'SELECT 1 FROM Players WHERE token=$1 AND game_id=$2'
+  return (await client.query(query, [playerToken, gameId])).rowCount !== 0
+}
+
+export async function isPlayersTurn(
+  playerToken: string,
+  gameId: string,
+  client: Client
+): Promise<boolean> {
+  const query = 'SELECT 1 FROM Games WHERE game_id=$1 AND current_player=$2'
+  return (await client.query(query, [gameId, playerToken])).rowCount !== 0
+}
+
 export async function deletePlayer(playerToken: string, client: Client) {
   const query = 'DELETE FROM Players WHERE token=$1'
   await client.query(query, [playerToken])
 }
 
+export async function setPlayerState(
+  playerToken: string,
+  client: Client,
+  state: string
+) {
+  const query = 'UPDATE Players SET last_action=$1 WHERE token=$2'
+  await client.query(query, [state, playerToken])
+}
+
+export async function getPlayerState(playerToken: string, client: Client) {
+  const query = 'SELECT last_action FROM Players WHERE player_token=$1'
+  await client.query(query, [playerToken])
+}
+
+export async function setNewCurrentPlayer(
```

Comment:
This function is quite long and complex, maybe you should divide it into other functions?